templates/service-template.yml

```diff
@@ -264,9 +264,14 @@ objects:
             env:
               - name: "AMS_ENV"
                 value: "${ENVIRONMENT}"
+              - name: POD_NAME
+                valueFrom:
+                  fieldRef:
+                    fieldPath: metadata.name
             command:
             - /usr/local/bin/maestro
             - server
+            - --client-id=$(POD_NAME)
```

Comment:
can we use a rand id here? we may need to consider how to use this client id do the authZ and authn with AWS IoT, right?

Reply:
Is it necessary to differentiate between different Maestro instances or pods connecting to AWS IoT? I checked the AWS IoT policy intended for Maestro, and it appears as follows:

```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": "iot:Connect",
      "Resource": "arn:aws:iot:<region>:<aws-accountID>:consumers/maestro-pub-consumer",
"arn:aws:iot:<region>:<aws-accountID>:consumers/maestro-sub-consumer"
    },

```

Reply:
In `iot:Connect` part, I remembered `consumers/maestro-sub-consume` is the mqtt client id, I think you may have a test for this,  we may need a  wildcard (`consumers/maestro-*-consumer`) for source client in the policy?

Reply:
it works. I tried to add a policy: (the last resource item is added, other are from aws sample)
```json
    {
      "Effect": "Allow",
      "Action": "iot:Connect",
      "Resource": [
        "arn:aws:iot:us-east-1:xxxxx:client/sdk-java",
        "arn:aws:iot:us-east-1:xxxxx:client/basicPubSub",
        "arn:aws:iot:us-east-1:xxxxx:client/sdk-nodejs-*",
        "arn:aws:iot:us-east-1:xxxxx:client/maestro-*-consumer"
      ]
    }
```

and send a message with the following comand:

```bash
$ mosquitto_pub -h xxxxxx-ats.iot.us-east-1.amazonaws.com --key lcao-thing-1.private.key --cert lcao-thing-1.cert.pem --cafile root-CA.crt -i maestro-test-consumer --topic sdk/test/js -m "test pub from maestro client."
```

Reply:
so we may set the client to `- --client-id=maestro-$(POD_NAME)` ? then we can have a policy  
```
"Resource": [
        "arn:aws:iot:us-east-1:xxxxx:client/maestro-*"
      ]
```